resolver/src/main/java/io/netty/resolver/RoundRobinInetAddressResolver.java

```diff
@@ -64,7 +63,7 @@ public void operationComplete(Future<List<InetAddress>> future) throws Exception
                     if (numAddresses > 0) {
                         // if there are multiple addresses: we shall pick one by one
                         // to support the round robin distribution
-                        promise.setSuccess(inetAddresses.get(index.getAndIncrement() % numAddresses));
+                        promise.setSuccess(inetAddresses.get(shiftSize(numAddresses)));
```

Comment:
@guidomedina another solution could "can cause stuck into static resolution in some setups." Scott even provided an example here https://github.com/netty/netty/commit/5eebe9a06c6873e797b5f8c847086c35ae990ef5#commitcomment-19776079 Better safe than sorry. And better ineffective solution than incorrect. Feel free to submit effective solution without this bug. I'm not ready right now.